backend/api/search.py

```diff
@@ -128,6 +128,101 @@ def search_api():
         return jsonify(projects)
 
 
+@search.route('/api/projects/search/count', methods=['POST'])
+@login_required
+def search_count():
+    """Search function
+
+    Returns:
+        res (json): JSON containing number of matches
+
+    """
+
+    query = request.get_json()
+
+    search_string = query.get('searchString', '')
+    archived = query.get('archived')
+    authors = query.get('authors')
+    tags = query.get('tags')
+    status = query.get('status')
+    date_from = query.get('date_from')
+    date_to = query.get('date_to')
+    description = query.get('description')
+    title = query.get('title')
+
+    request_json = {
+        'query': {
+            'bool': {
+                'should': {
+                    'multi_match': {
+                        'query': search_string,
+                        'fields': [
+                            'tags^2',
+                            'title^2',
+                            'description',
+                        ],
+                    }
+                },
+                'filter': {
+                    'bool': {
+                        'must': []
+                    }
+                }
+            }
+        }
+    }
+
+    if authors:
+        for author in authors:
+            request_json['query']['bool']['filter']['bool']['must'].append({
+                'term': {'authors': author}
+                })
+
+    if tags:
+        for tag in tags:
+            request_json['query']['bool']['filter']['bool']['must'].append({
+                'term': {'tags': tag}
+            })
+
+    if status:
+        request_json['query']['bool']['filter']['bool']['must'].append({
+            'term': {'status': status.lower()}
+        })
+
+    if archived:
+        request_json['query']['bool']['filter']['bool']['must'].append({
+            'term': {'archived': archived in ['true', 'True', '1', 'yes', 'y']}
+        })
+
+    if date_from or date_to:
+        date_filter = {
+            'range': {'date_creation': {}}
+        }
+
+        if date_from:
+            date_filter['range']['date_creation']['gte'] = date_from
+        if date_to:
+            date_filter['range']['date_creation']['lte'] = date_to
+
+        request_json['query']['bool']['filter']['bool']['must'].append(date_filter)
+
+    if description:
+        request_json['query']['bool']['filter']['bool']['must'].append({
+            'wildcard': {'description': '*' + description + '*'}
+        })
+
+    if title:
+        request_json['query']['bool']['filter']['bool']['must'].append({
+            'wildcard': {'title': '*' + title + '*'}
+        })
+
+    try:
+        res = g.es.search(index="knexdb", body=request_json)['hits']['total']
+        return jsonify(res)
+    except Exception as err:
+        raise ApiException(str(err), 400)
+
+
```

Comment:
This amount of duplicated code is hideous as fuck.
Please make a new `es_query` function or something like that out of it, where you just stuff in `query` and get the final `request_json`.